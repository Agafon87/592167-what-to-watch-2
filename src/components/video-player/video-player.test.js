import React from "react";
import renderer from "react-test-renderer";
import VideoPlayer from "./video-player.jsx";

const mock = {
  className: ``,
  image: ``,
  video: ``,
  isPlaying: true,
  isFull: true,
  size: {},
  onLoad: jest.fn(),
  onUpdateTime: jest.fn(),
};

describe(`VideoPlayer component`, () => {
  const {
    className,
    image,
    video,
    isPlaying,
    isFull,
    size,
    onLoad,
    onUpdateTime,
  } = mock;

  it(`renders correctly`, () => {
    const tree = renderer.create(<VideoPlayer
      className={className}
      image={image}
      video={video}
      isPlaying={isPlaying}
      isFull={isFull}
      size={size}
      onLoad={onLoad}
      onUpdateTime={onUpdateTime}
    />,
    {createNodeMock: (el) => {
      return el;
    }}).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
