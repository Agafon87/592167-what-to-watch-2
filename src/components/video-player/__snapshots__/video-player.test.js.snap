// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayer correctly renders after test 1`] = `
<video
  height={175}
  onMouseEnter={[Function]}
  width={280}
/>
`;
